sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2393,6 +2395,40 @@ def expand(self, pcoll):
           'The method to read from BigQuery must be either EXPORT'
           'or DIRECT_READ.')
 
+  def produce_pcoll_with_schema(self, project_id, dataset_id, table_id):
+    the_table_schema = beam.io.gcp.bigquery.bigquery_tools.BigQueryWrapper(
+    ).get_table(project_id, dataset_id, table_id)
+    the_schema = beam.io.gcp.bigquery_tools.get_dict_table_schema(
+        the_table_schema)
+    i = 0
+    dict_of_tuples = []
+    for x in the_schema['fields']:
+      if the_schema['fields'][i]['type'] == 'STRING':
+        typ = str
+      elif the_schema['fields'][i]['type'] == 'INTEGER':
+        typ = np.int64
+      elif the_schema['fields'][i]['type'] == 'FLOAT':
+        typ = np.float64
+      elif the_schema['fields'][i]['type'] == 'NUMERIC':
+        typ = np.float128
+      elif the_schema['fields'][i]['type'] == 'BIGNUMERIC':
+        typ = np.float128
```

Comment:
I think we should hold off defining mappings for these, np.float128 isn't a perfect mapping since it's floating point and these are fixed point types.